### ISSUE TITLE
bcnn_finetuning.py run error

### ISSUE THREAD

**Issue by Hedlen** [abhaydoke09/Bilinear-CNN-TensorFlow#9]
When I run the bcnn_finetuning.py,and  the following error occurs.My category is 134 classes.

`InvalidArgumentError (see above for traceback): logits and labels must be same size: logits_size=[4,134] labels_size=[16,134] [[Node: SoftmaxCrossEntropyWithLogitsSoftmaxCrossEntropyWithLogits[T=DT_FLOAT, _device="/job:loc
alhost/replica:0/task:0/gpu:0"](Reshape_2, Reshape_3)]]`
**logits_size and labels_size are not equal.**
I don't know why this happens. I'm running exactly as you follow your code.
Please help me solve this problem. 
Thank you very much!

**Comment by Hedlen** [comment]
I have understood, I was wrong. Thank you very much.

**Comment by rnsandeep** [comment]
Same Issue with me. I changed images to 224 instead of 448. 
Resolved it by changing this line in bcnn_finetuning.py
218.        self.conv5_3 = tf.reshape(self.conv5_3,[-1,512,784])            ''' Reshape conv5_3 from [batch_size, number_of_filters, height*width]
to  self.conv5_3 = tf.reshape(self.conv5_3,[-1,512,196]) # 784 depends on the size of image i guess. so for 224, 196 is correct number.
I think instead of hard coding 784 better to get it from tensors.

**Comment by Hedlen** [comment]
@rnsandeep ,Thank you for your replay. I will follow your advice. Thanks again!

**Comment by YanShuo1992** [comment]
@rnsandeep @Hedlen Hi, I also meet the same issue and your answers solve it.  However, I still have two questions:
1. Do you I need to change the line 232: self.phi_I = tf.divide(self.phi_I,784.0) ?
2. How to calculate the correct value if I use the images in other size?

Cheers

**Comment by rnsandeep** [comment]
@YanShuo1992  I can tell for multiples of 224. if you make your image 112  i.e. 224/2 it should be 49( 196/4). if you make it 448 (224x2) it should be 196*4( 784). Not sure about every size.

**Comment by YanShuo1992** [comment]
@rnsandeep my image is 720*720.  I find you can output the shape of the conv5_3.  It is (?, 512,14,14) when the image size is 224.  You can simply store the shape of the conv5_3 as s and replace the 196 with s*s.

**Comment by xiangzhang06** [comment]
logits and labels must be same size: logits_size=[4,200] labels_size=[16,200]. How did you solve it?

**Comment by xiangzhang06** [comment]
logits and labels must be same size: logits_size=[4,200] labels_size=[16,200]. How did you solve it?

**Comment by xiangzhang06** [comment]
logits and labels must be same size: logits_size=[4,200] labels_size=[16,200]. How did you solve it? @Hedlen

**Comment by jiayugedede** [comment]
In bcnn_finetuning.py--> for i in range(total_val_batch): <-- have occured error.It's undefine name.
